shell/components/form/LabeledSelect/labeled-select-pagination.ts

```diff
@@ -0,0 +1,134 @@
+import { LABEL_SELECT_NOT_OPTION_KINDS } from './LabeledSelect.utils';
+import { debounce } from 'lodash';
+import Vue, { PropType } from 'vue';
+
+interface Data {
+  currentPage: number,
+  search: string,
+  pageSize: number,
+
+  page: any[],
+  pages: number,
+  totalResults: number,
+
+  paginating: boolean,
+
+  debouncedRequestPagination: Function
+}
+
+interface Methods {
+  loadMore: () => void
+  setPaginationFilter: (filter: string) => void
+  requestPagination: () => Promise<any>;
+}
+
+/**
+ * Make a http request to fetch paginated results
+ *
+ * @param [array] pageContent Current page
+ * @param [number] page page number to fetch
+ * @param [number] pageSize number of items in the page to fetch
+ * @param [string] filter pagination filter. this is just a text string associated with user entered text
+ * @param [boolean] resetPage true if the result should only contain the fetched page, false if the result should be added to the pageContent
+ */
+type PaginateFn<T = any> = (opts: {
+  pageContent: T[],
+  page: number,
+  pageSize: number,
+  filter: string,
+  resetPage: boolean,
+}) => Promise<{
+  page: T[],
+  pages: number,
+  total: number
+}>
+
+interface Props {
+  paginate: PaginateFn
+}
+
+export default Vue.extend<Data, Methods, any, Props>({
```

Comment:
Shouldn't we be moving away of `Vue.extend` as per https://github.com/rancher/dashboard/issues/10104?